### PR TITLE
DO NOT MERGE: A test for change only linting

### PR DESCRIPTION
**Pull request by blowery**
A test for change only linting.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ jobs:
       - run:
           name: Lint Client and Server
           command: |
-            ./node_modules/.bin/eslint-eslines $(circleci tests glob "client/**/*.js" "client/**/*.jsx" "server/**/*.js" "server/**/*.jsx" | circleci tests split)
+            ./node_modules/.bin/eslint-eslines $(git diff --name-only origin/master... | grep -E "^(client|server).+\.jsx?" | circleci tests split)
```
Comment (sirreal) [inline .circleci/config.yml]: How about grepping "starts with" and "ends with" separately:

```sh
grep -E '^(client|server)/' | grep -E '\.jsx?$'
```

**Comment by sirreal** [comment]
02:53 after d4281c3 😮

**Comment by blowery** [comment]
Something's wrong. Linting two files shouldn't take ~3m without splitting.

**Comment by sirreal** [comment]
Right, I don't think we should _need_ to split. Most PRs aren't touching that many files. See https://github.com/Automattic/wp-calypso/pull/25494#issuecomment-397359925

> Upsetting that this step still takes 00:36 to lint no files (!! how can that be !!), but much better than last master build at 01:33.

**Comment by sirreal** [comment]
0:01 🎉 ❤️ 🎉 👏 👏 👏 👏

**Comment by blowery** [comment]
updated #25494, closing this

**Comment by blowery** [comment]
> Right, I don't think we should need to split.

Sure, though the split does isolate the failure to one runner instead of all of them.

**Pull request closed without merging**